// @fairfx/geometry-web/dist/theme/tokens.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tokens = exports.transitions = exports.breakpoints = exports.zIndices = exports.sizes = exports.spaces = void 0;
exports.spaces = {
    px: '1px',
    '0': '0',
    '1': '0.25rem',
    '2': '0.5rem',
    '3': '0.75rem',
    '4': '1rem',
    '5': '1.25rem',
    '6': '1.5rem',
    '8': '2rem',
    '10': '2.5rem',
    '12': '3rem',
    '16': '4rem',
    '20': '5rem',
    '24': '6rem',
    '32': '8rem',
    '40': '10rem',
    '48': '12rem',
    '56': '14rem',
    '64': '16rem',
};
exports.sizes = Object.assign(Object.assign({}, exports.spaces), { full: '100%', half: '50%', '3xs': '14rem', '2xs': '16rem', xs: '20rem', sm: '24rem', md: '28rem', lg: '32rem', xl: '36rem', '2xl': '42rem', '3xl': '48rem', '4xl': '56rem', '5xl': '64rem', '6xl': '72rem' });
exports.zIndices = {
    hide: -1,
    auto: 'auto',
    base: 0,
    docked: 10,
    dropdown: 1000,
    sticky: 1100,
    banner: 1200,
    overlay: 1300,
    popover: 1500,
    skipLink: 1600,
    toast: 1700,
    tooltip: 1800,
    modal: 9999,
};
exports.breakpoints = {
    xs: '36rem',
    sm: '48rem',
    md: '62rem',
    lg: '87.5rem',
    xl: '100rem',
};
exports.transitions = {
    base: '300ms',
    fast: '200ms',
};
exports.tokens = {
    transitions: exports.transitions,
    breakpoints: exports.breakpoints,
    space: exports.spaces,
    sizes: exports.sizes,
    zIndices: exports.zIndices,
};
