// @fairfx/geometry-web/dist/components/Radio/Radio.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RadioInnerCircle = exports.RadioWrapper = void 0;
const core_1 = require("@emotion/core");
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
exports.RadioWrapper = theme_1.default.label `
  display: block;
  position: relative;
  padding-left: ${tokens_1.spaces[9]}px;
  margin-bottom: ${tokens_1.spaces[2]}px;
  margin-top: ${tokens_1.spaces[7]}px;
  ${({ isDisabled }) => !isDisabled &&
    core_1.css `
      cursor: pointer;
    `}
  font-size: ${tokens_1.size.font.small.value}px;
  user-select: none;
  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    &:checked {
      ~ span {
        background-color: ${({ isDisabled, theme }) => isDisabled && `${theme.palette.white[0]}`};
        border-radius: 50%;
        border: 2px solid ${({ isDisabled, theme }) => isDisabled
    ? `${theme.palette.grey[2]}`
    : `${theme.palette.midnightBlue[0]}`};
      }
    }
  }
  &:hover {
    span {
      ${({ isDisabled, theme }) => !isDisabled &&
    core_1.css `
          background-color: ${theme.palette.grey[2]};
        `};
    }
  }
`;
exports.RadioInnerCircle = theme_1.default.span `
  position: absolute;
  top: -2px;
  left: 0;
  height: 24px;
  width: 24px;
  background-color: ${p => p.theme.palette.white[0]};
  border-radius: 50%;
  border: 2px solid
    ${({ isDisabled, theme }) => isDisabled
    ? `${theme.palette.grey[2]}`
    : `${theme.palette.midnightBlue[0]}`};
  &:after {
    content: '';
    position: absolute;
    display: ${props => (props.checked ? 'block' : 'none')};
    top: 3px;
    left: 3px;
    width: 14px;
    height: 14px;
    border-radius: 50%;
    background-color: ${({ isDisabled, theme }) => isDisabled
    ? `${theme.palette.grey[2]}`
    : `${theme.palette.midnightBlue[0]}`};
  }
`;
