// @fairfx/geometry-web/dist/components/tooltip/Tooltip.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TooltipContent = exports.TooltipRoot = exports.Text = void 0;
const core_1 = require("@emotion/core");
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
const TooltipContentStyles = props => core_1.css `
  border-style: solid;
  border-color: ${props.theme.palette.midnightBlue[0]} transparent transparent
    transparent;
  border-width: ${tokens_1.spaces[5]}px ${tokens_1.spaces[5]}px 0 ${tokens_1.spaces[5]}px;
  content: ' ';
  height: 0;
  width: 0;
  position: absolute;
  pointer-events: none;
  top: 89%;
`;
exports.Text = theme_1.default.span `
  box-sizing: border-box;
  overflow: hidden;
  padding: ${tokens_1.spaces[4]}px ${tokens_1.spaces[7]}px;
  font-weight: bold;
  font-size: ${tokens_1.fontSizes[1]}px;
  color: ${p => p.theme.palette.white[0]};
  user-select: none;
  border-radius: ${tokens_1.spaces[2]}px;
  display: inline-block;
  background-color: ${p => p.theme.palette.midnightBlue[0]};
`;
exports.TooltipRoot = theme_1.default('div') `
  position: relative;
  display: inline-block;
`;
exports.TooltipContent = theme_1.default.div `
  position: absolute;
  text-align: left;
  z-index: ${tokens_1.indices.tooltip};
  bottom: calc(100% + 10px);
  white-space: pre-wrap;
  /**Centre alignment */
  width: 300px; /**Some really big number that we won't realistically hit but it gets all widths in absolute the same */
  left: -150px; /**Centre the width */
  margin-left: 50%;
  text-align: center;

  ${props => props.align === 'center'
    ? core_1.css `
          &:before {
            left: 50%;
            margin-left: -${tokens_1.spaces[5]}px;
            ${TooltipContentStyles(props)}
          }
        `
    : core_1.css `
          width: 0;
          left: 0;
          margin-left: 0;
          text-align: left;

          &:before {
            margin-left: ${tokens_1.spaces[5]}px;
            ${TooltipContentStyles(props)}
          }
        `}
`;
