// @fairfx/geometry-web/dist/components/logos/LogoEquals.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = exports.Equals = exports.TextWrapper = exports.EqualsWrapper = exports.LogoEqualsRoot = void 0;
const is_prop_valid_1 = __importDefault(require("@emotion/is-prop-valid"));
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
exports.LogoEqualsRoot = theme_1.default.div `
  display: flex;
  align-items: center;
  justify-content: flex-start;
  height: ${p => p.height}px;
  overflow: hidden;
`;
exports.EqualsWrapper = theme_1.default.div `
  height: ${tokens_1.sizes[10]}px;
  width: ${tokens_1.sizes[10]}px;
  margin-left: ${tokens_1.sizes[3]}px;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
`;
exports.TextWrapper = theme_1.default.div `
  height: ${tokens_1.sizes[10]}px;
  display: flex;
  align-items: center;
  flex-shrink: 0;
`;
exports.Equals = theme_1.default('path', {
    shouldForwardProp: is_prop_valid_1.default,
}) `
  fill: ${p => (p.fill ? p.fill : tokens_1.color.charcoal.base.value)};
`;
exports.Text = theme_1.default('path', {
    shouldForwardProp: is_prop_valid_1.default,
}) `
  fill: ${p => (p.fill ? p.fill : tokens_1.color.charcoal.base.value)};
`;
