// @fairfx/geometry-web/dist/components/header/Header.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceTitle = exports.PageTitle = exports.NavBar = exports.LeftItems = exports.HeaderRoot = void 0;
const core_1 = require("@emotion/core");
const is_prop_valid_1 = __importDefault(require("@emotion/is-prop-valid"));
const mediaQuery_1 = __importDefault(require("../../common/mediaQuery"));
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
const Text_1 = require("../typography/Text");
const SIZES = {
    height: '56px',
};
exports.HeaderRoot = theme_1.default.div `
  position: relative;
  height: ${SIZES.height};
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-top: 0;
  margin-bottom: 0;
  box-shadow: 'none';

  ${mediaQuery_1.default.tablet} {
    box-shadow: 0 1px 0 0 ${tokens_1.color.black_haze.base.value};
  }
`;
exports.LeftItems = theme_1.default.div `
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  display: flex;
  align-items: 'center';
`;
exports.NavBar = theme_1.default('div', {
    shouldForwardProp: is_prop_valid_1.default,
}) `
  display: flex;
  padding: 20px;
  cursor: pointer;
  ${(props) => props.isCollapsed
    ? core_1.css `
          background: ${tokens_1.color.white.base.value};
        `
    : ''}
  ${mediaQuery_1.default.desktop} {
    display: none;
  }
`;
exports.PageTitle = theme_1.default(Text_1.Text) `
  ${mediaQuery_1.default.desktop} {
    display: none;
    margin-left: 7px;
    user-select: none;
  }
`;
exports.BalanceTitle = theme_1.default(Text_1.Text) `
  display: none;
  margin-right: 6px;
  margin-left: 6px;
  user-select: 'none';
  ${mediaQuery_1.default.desktop} {
    display: flex;
  }
`;
