// @fairfx/geometry-web/dist/components/CardStatus/CardStatus.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const Box_1 = __importDefault(require("../Box"));
const IconCardCancelled_1 = __importDefault(require("../icons/IconCardCancelled"));
const IconCardLostOrStolen_1 = __importDefault(require("../icons/IconCardLostOrStolen"));
const IconCardNeedsActivation_1 = __importDefault(require("../icons/IconCardNeedsActivation"));
const IconCardPaused_1 = __importDefault(require("../icons/IconCardPaused"));
const IconCards_1 = __importDefault(require("../icons/IconCards"));
const Inline_1 = __importDefault(require("../layout/Inline"));
const Text_1 = __importDefault(require("../typography/Text"));
const mapUserStatus = (status) => ({
    active: react_1.default.createElement(IconCards_1.default, null),
    ordered: react_1.default.createElement(IconCards_1.default, null),
    needs_activation: react_1.default.createElement(IconCardNeedsActivation_1.default, null),
    paused: react_1.default.createElement(IconCardPaused_1.default, null),
    lost: react_1.default.createElement(IconCardLostOrStolen_1.default, null),
    stolen: react_1.default.createElement(IconCardLostOrStolen_1.default, null),
    fraudulent: react_1.default.createElement(IconCardLostOrStolen_1.default, null),
    cancelled: react_1.default.createElement(IconCardCancelled_1.default, null),
    unknown: null,
}[status]);
const changeTextColor = (status) => {
    switch (status) {
        case 'lost':
        case 'stolen':
        case 'fraudulent':
        case 'cancelled':
            return 'danger';
        default:
            return 'base';
    }
};
const CardStatus = ({ cardStatus, cardStatusText, className, revert = false, }) => {
    return (react_1.default.createElement(Inline_1.default, { className: className },
        react_1.default.createElement(Box_1.default, { marginRight: "1", opacity: revert ? 0.5 : 1 }, mapUserStatus(cardStatus)),
        react_1.default.createElement(Text_1.default, { color: revert ? 'muted' : changeTextColor(cardStatus), as: "div", fontSize: "small", "data-testid": "cardStatus_text" }, cardStatusText)));
};
exports.default = CardStatus;
