// @fairfx/geometry-web/dist/components/tables/Table.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableCellRoot = exports.TableRowRoot = exports.TableHeadingRoot = exports.TableRoot = void 0;
const core_1 = require("@emotion/core");
const mediaQuery_1 = __importDefault(require("../../common/mediaQuery"));
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
const utils_1 = require("../../themer/utils");
exports.TableRoot = theme_1.default('table') `
  width: 100%;
  border-collapse: collapse;
`;
exports.TableHeadingRoot = theme_1.default('th') `
  vertical-align: middle;
  padding: ${utils_1.spacing('times0p75')};
  position: sticky;
  background-color: ${tokens_1.color.manatee.base.value};
  color: ${tokens_1.color.white.base.value};
  top: ${utils_1.sizing('headerAction')};
  z-index: ${tokens_1.indices.listSubheader};

  &:nth-of-type(1) {
    padding-left: 32px;
  }

  &:nth-last-of-type(1) {
    padding-right: 32px;
  }

  ${mediaQuery_1.default.desktop} {
    top: ${utils_1.sizing('action')};
  }
`;
const chevronCharcoal = core_1.css `
  background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNyIgaGVpZ2h0PSIxMiIgdmlld0JveD0iMCAwIDcgMTIiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNMS4zMTYxOSAwTDcgNkwxLjMxNjE5IDEyTDAgMTAuNjEwNkw0LjM2NzYyIDZMMCAxLjM4OTQxTDEuMzE2MTkgMFoiIGZpbGw9IiMzMDMzNDQiLz4KPC9zdmc+Cg==');
`;
exports.TableRowRoot = theme_1.default('tr') `
  transition: background-color ${tokens_1.transition.base};
  position: relative;
  box-shadow: ${tokens_1.color.black_haze.base.value} 0px -1px 0px 0px;
  text-decoration: none;

  background-color: ${({ isSelectable, isSelected }) => isSelectable && isSelected
    ? tokens_1.color.gallery.base.value
    : tokens_1.color.white.base.value};

  &:hover {
    background-color: ${tokens_1.color.gallery.base.value};
    cursor: ${({ isSelectable }) => (isSelectable ? 'pointer' : undefined)};
  }

  .TableRow_TableCell:nth-of-type(1) {
    padding-left: 32px;
  }

  .TableRow_TableCell:nth-last-of-type(1) {
    padding-right: 32px;
  }

  ${({ isSelectable }) => isSelectable &&
    core_1.css `
      &:after {
        content: '';
        display: block;
        position: absolute;
        width: 7px;
        height: 12px;
        margin-left: -17px;
        margin-top: 26px;
        background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNyIgaGVpZ2h0PSIxMiIgdmlld0JveD0iMCAwIDcgMTIiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNMS4zMTYxOSAwTDcgNkwxLjMxNjE5IDEyTDAgMTAuNjEwNkw0LjM2NzYyIDZMMCAxLjM4OTQxTDEuMzE2MTkgMFoiIGZpbGw9IiNBRUJDQ0YiLz4KPC9zdmc+Cg==');
      }

      &:hover:after {
        ${chevronCharcoal}
      }
    `}

  ${({ isSelected }) => isSelected &&
    core_1.css `
      &:after {
        ${chevronCharcoal}
      }

      .TableRow_TableCell:nth-of-type(1) {
        box-shadow: inset ${tokens_1.color.mango.base.value} 3px 0px 0px 0px;
      }
    `}
`;
exports.TableCellRoot = theme_1.default('td') `
  padding: ${utils_1.spacing('small')};
  height: ${utils_1.sizing('action')};
  min-height: ${utils_1.sizing('action')};
  width: ${({ width }) => (utils_1.isSpacingValue(width) ? utils_1.spacing(width) : width)};
  vertical-align: ${({ vAlign = 'middle' }) => vAlign};
`;
