// @fairfx/geometry-web/dist/components/layout/Layout.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InlineRoot = exports.BoxRoot = void 0;
const is_prop_valid_1 = __importDefault(require("@emotion/is-prop-valid"));
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
const utils_1 = require("../../themer/utils");
/**
 * @deprecated Use XBox component instead
 */
exports.BoxRoot = theme_1.default('div', {
    shouldForwardProp: prop => is_prop_valid_1.default(prop) && !['display', 'width'].includes(prop),
}) `
  background-color: ${({ bg = 'none' }) => ({
    none: 'transparent',
    base: 'transparent',
    first: tokens_1.color.white.base.value,
    second: tokens_1.color.charcoal.base.value,
    third: tokens_1.color.mango.base.value,
    light: tokens_1.color.white.base.value,
    dark: tokens_1.color.charcoal.base.value,
    brand: tokens_1.color.mango.base.value,
}[bg])};

  padding: ${({ p = 'zero' }) => utils_1.spacing(p)};
  padding-top: ${({ pt = 'zero' }) => utils_1.spacing(pt)};
  padding-left: ${({ pl = 'zero' }) => utils_1.spacing(pl)};
  padding-right: ${({ pr = 'zero' }) => utils_1.spacing(pr)};
  padding-bottom: ${({ pb = 'zero' }) => utils_1.spacing(pb)};

  box-shadow: ${({ shadow = 'none' }) => ({
    base: 'none',
    none: 'none',
    shallow: `2px 2px 1px ${tokens_1.color.black_haze.base.value}`,
}[shadow])};

  /** Experimental */
  display: ${({ display = '' }) => ({
    '': undefined,
    flex: 'flex',
    inline: 'inline-flex',
}[display])};
  justify-content: ${({ alignContentX = 'left' }) => ({
    left: 'flex-start',
    center: 'center',
    right: 'flex-end',
    spaceBetween: 'space-between',
}[alignContentX])};
  align-items: ${({ alignContentY = 'top' }) => ({
    top: 'flex-start',
    center: 'center',
    bottom: 'flex-end',
}[alignContentY])};
  width: ${({ width = '' }) => ({
    '': undefined,
    full: '100%',
}[width])};
`;
exports.InlineRoot = theme_1.default(exports.BoxRoot) `
  display: inline-flex;
`;
