// @fairfx/geometry-web/dist/components/forms/PasswordPanel/PasswordPanel.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Title = exports.Text = exports.Line = exports.Icon = exports.Criteria = exports.PasswordPanelRoot = void 0;
const is_prop_valid_1 = __importDefault(require("@emotion/is-prop-valid"));
const theme_1 = __importDefault(require("../../../theme"));
exports.PasswordPanelRoot = theme_1.default.div `
  border-radius: 5px;
  padding: 10px 20px;
  margin-top: 20px;
  line-height: 0.8;
`;
exports.Criteria = theme_1.default.div `
  display: flex;
  align-items: center;
  padding-bottom: 20px;
`;
exports.Icon = theme_1.default.div `
  width: 16px;
  height: 14px;
  margin: 0 5px;
  display: flex;
  align-items: center;
  justify-content: center;
  svg {
    width: 16px;
  }
`;
exports.Line = theme_1.default.div `
  width: 10px;
  height: 1px;
  background-color: ${p => p.theme.palette.grey[5]};
`;
exports.Text = theme_1.default('span', {
    shouldForwardProp: is_prop_valid_1.default,
}) `
  color: ${p => p.isChecked ? p.theme.palette.midnightBlue[0] : p.theme.palette.grey[5]};
`;
exports.Title = theme_1.default.p `
  color: ${p => p.theme.palette.midnightBlue[0]};
  padding-bottom: 10px;
`;
