// @fairfx/geometry-web/dist/components/PrimitiveModal/PrimitiveModal.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionModalWrapper = exports.CloseButtonWrapper = exports.ModalRoot = exports.Backdrop = void 0;
const rgba_1 = __importDefault(require("polished/lib/color/rgba"));
const theme_1 = __importDefault(require("../../theme"));
exports.Backdrop = theme_1.default.div `
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${p => p.backdropVisible
    ? rgba_1.default(p.theme.palette.midnightBlue[0], 0.5)
    : 'transparent'};
  z-index: ${p => p.theme.zIndices.modal};
`;
exports.ModalRoot = theme_1.default.div `
  position: relative;
  background-color: ${p => p.theme.palette.white[0]};
  padding: ${p => p.theme.space[p.padding]};
  min-width: ${p => p.theme.sizes[p.minWidth]};
  min-height: ${p => p.theme.sizes[p.minHeight]};
  max-height: 95vh;
  overflow: ${p => (p.overflow ? p.overflow : 'auto')};
  max-width: 95vw;
`;
exports.CloseButtonWrapper = theme_1.default.div `
  position: absolute;
  top: ${p => p.theme.space[6]};
  right: ${p => p.theme.space[6]};
`;
exports.ActionModalWrapper = theme_1.default.div `
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;
