// @fairfx/geometry-web/dist/components/Checkbox/Checkbox.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Label = exports.CheckboxContainer = exports.StyledCheckbox = exports.CheckMarkIcon = exports.HiddenCheckbox = exports.Wrapper = void 0;
const theme_1 = __importDefault(require("../../theme"));
exports.Wrapper = theme_1.default.label `
  display: flex;
`;
exports.HiddenCheckbox = theme_1.default.input `
  border: 0;

  height: 1px;
  width: 1px;

  margin: -1px;
  padding: 0;

  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  outline: none;
  opacity: 0;
`;
exports.CheckMarkIcon = theme_1.default.svg `
  fill: none;
  stroke: white;
  stroke-width: 3px;
`;
exports.StyledCheckbox = theme_1.default.div `
  display: inline-block;
  width: ${p => p.theme.space[5]};
  height: ${p => p.theme.space[5]};
  background: ${p => p.checked ? p.theme.palette.midnightBlue[0] : p.theme.palette.white[0]};
  border-radius: 3px;
  transition: all 150ms;
  box-shadow: inset 0 0 0 2px ${p => p.theme.palette.midnightBlue[0]};

  .check-mark-icon {
    visibility: ${p => (p.checked ? 'visible' : 'hidden')};
    padding-bottom: 1px;
  }

  ${p => p.focused &&
    `
    box-shadow: inset 0 0 0 2px ${p.theme.palette.midnightBlue[0]}, 0 0 0 2px  ${p.theme.palette.skyBlue[2]};
  `};
`;
exports.CheckboxContainer = theme_1.default.div `
  margin-top: ${p => p.theme.space[1]};
  display: inline-block;
  vertical-align: middle;
`;
exports.Label = theme_1.default.span `
  margin-left: ${p => p.theme.space[3]};
  font-size: 16px;
  line-height: 24px;
  color: ${p => p.theme.palette.grey[7]};
`;
