// @fairfx/geometry-web/dist/components/forms/InputWithAction/InputWithAction.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Action = exports.Input = exports.InputWithActionRoot = void 0;
const is_prop_valid_1 = __importDefault(require("@emotion/is-prop-valid"));
const theme_1 = __importDefault(require("../../../theme"));
const transitions_1 = __importDefault(require("../../../themer/base/transitions"));
exports.InputWithActionRoot = theme_1.default.div `
  position: relative;
`;
exports.Input = theme_1.default('div', {
    shouldForwardProp: is_prop_valid_1.default,
}) `
  position: relative;
  z-index: 1;
  transition: all ${transitions_1.default.base};
  width: ${props => props.width || '100%'};
`;
exports.Action = theme_1.default('div', {
    shouldForwardProp: is_prop_valid_1.default,
}) `
  position: absolute;
  top: 0;
  right: 0;
  > * {
    min-width: 128px;
  }
  transition: all ${transitions_1.default.base};
  opacity: ${props => (props.isFilled ? 1 : 0)};
`;
