// @fairfx/geometry-web/dist/themer/base/colors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.lightModeColors = void 0;
const tokens_1 = require("../tokens");
exports.lightModeColors = {
    boxShadow: '#00000021',
    base: tokens_1.color.charcoal.base.value,
    alt: tokens_1.color.white.base.value,
    danger: tokens_1.color.danger.base.value,
    accentColorOne: tokens_1.color.mango.base.value,
    accentColorTwo: tokens_1.color.teal.base.value,
    primary: tokens_1.color.purple.base.value,
    primaryAlt: tokens_1.color.purple.dark.value,
    secondary: tokens_1.color.tuna.base.value,
    secondaryAlt: tokens_1.color.mischka.base.value,
    muted: tokens_1.color.athens_gray.light.value,
    mutedAlt: tokens_1.color.casper.base.value,
    pageBackground: tokens_1.color.athens_gray.base.value,
    fieldDefault: tokens_1.color.athens_gray.base.value,
    fieldFocus: tokens_1.color.white_lilac.base.value,
    seperator: tokens_1.color.black_haze.base.value,
    brand: tokens_1.color.mango.base.value,
    lightNavigationSelectedBackground: tokens_1.color.silver.base.value,
    tableHeader: tokens_1.color.manatee.base.value,
    listRowBackgroundHover: tokens_1.color.gallery.base.value,
    shadow: {
        base: tokens_1.color.black_haze.base.value,
    },
    fill: {
        BalanceSummaryLink: {
            base: tokens_1.color.purple.base.value,
            hover: tokens_1.color.white.base.value,
        },
        InspectorPanel: {
            Icon: {
                base: tokens_1.color.purple.base.value,
                hover: tokens_1.color.white.base.value,
                first: tokens_1.color.danger.base.value,
            },
            Card: {
                base: tokens_1.color.athens_gray.base.value,
                first: tokens_1.color.black_haze.base.value,
                second: tokens_1.color.purple.base.value,
            },
        },
        LogoEquals: {
            equals: {
                base: tokens_1.color.mango.base.value,
                first: tokens_1.color.mango.base.value,
                second: tokens_1.color.mango.base.value,
                third: tokens_1.color.charcoal.base.value,
            },
            text: {
                base: tokens_1.color.charcoal.base.value,
                first: tokens_1.color.charcoal.base.value,
                second: tokens_1.color.white.base.value,
                third: tokens_1.color.charcoal.base.value,
            },
        },
        Header: {
            text: {
                base: tokens_1.color.casper.base.value,
                dark: tokens_1.color.casper.dark.value,
            },
        },
        RadioButton: {
            base: tokens_1.color.purple.base.value,
            background: tokens_1.color.white.base.value,
            hover: tokens_1.color.purple.muted.value,
        },
        Dropdown: {
            base: tokens_1.color.casper.base.value,
            active: tokens_1.color.purple.base.value,
            Line: {
                base: tokens_1.color.mango.base.value,
            },
            Collapsable: {
                background: tokens_1.color.athens_gray.base.value,
            },
        },
        NavigationItem: {
            hover: tokens_1.color.purple.base.value,
        },
        Select: {
            Chevron: {
                base: tokens_1.color.purple.base.value,
            },
        },
        SideBarNavigation: {
            Icon: {
                base: tokens_1.color.white.base.value,
            },
        },
    },
    text: {
        base: tokens_1.color.charcoal.base.value,
        BalanceSummaryLink: {
            base: tokens_1.color.purple.base.value,
            hover: tokens_1.color.white.base.value,
        },
        DownloadPage: {
            base: tokens_1.color.lighter.base.value,
            title: tokens_1.color.charcoal.base.value,
        },
        ListCell: {
            highlight: tokens_1.color.purple.base.value,
        },
        NavigationItem: {
            base: tokens_1.color.charcoal.base.value,
            hover: tokens_1.color.purple.base.value,
        },
        Input: {
            base: tokens_1.color.charcoal.base.value,
            placeholder: {
                base: tokens_1.color.manatee.base.value,
                focus: tokens_1.color.purple.base.value,
            },
        },
        SelectorOption: {
            base: tokens_1.color.purple.base.value,
            active: tokens_1.color.white.base.value,
        },
        Tabs: {
            base: tokens_1.color.charcoal.base.value,
        },
        Text: {
            base: tokens_1.color.charcoal.base.value,
            light: tokens_1.color.white.base.value,
            disable: tokens_1.color.waterloo.base.value,
            muted: tokens_1.color.athens_gray.light.value,
            danger: tokens_1.color.danger.base.value,
        },
        Select: {
            label: {
                base: tokens_1.color.waterloo.base.value,
                active: tokens_1.color.purple.base.value,
            },
            base: tokens_1.color.charcoal.base.value,
            option: tokens_1.color.purple.base.value,
        },
    },
    background: {
        base: 'transparent',
        page: tokens_1.color.athens_gray.base.value,
        field: {
            base: tokens_1.color.athens_gray.base.value,
            focus: tokens_1.color.white_lilac.base.value,
            error: tokens_1.color.danger.light.value,
        },
        BalanceSummary: {
            divider: tokens_1.color.charcoal.base.value,
        },
        BalanceSummaryLink: {
            base: tokens_1.color.white.base.value,
            hover: tokens_1.color.casper.base.value,
        },
        Box: {
            base: 'transparent',
            first: tokens_1.color.white.base.value,
            second: tokens_1.color.charcoal.base.value,
            third: tokens_1.color.mango.base.value,
        },
        DashboardLayout: {
            base: tokens_1.color.athens_gray.base.value,
            headline: tokens_1.color.charcoal.base.value,
        },
        NavigationItem: {
            base: tokens_1.color.white.base.value,
            hover: tokens_1.color.purple.base.value,
            select: tokens_1.color.mango.base.value,
            selectBg: tokens_1.color.silver.base.value,
        },
        InformationBadge: {
            base: tokens_1.color.silver.base.value,
        },
        Inline: {
            base: 'transparent',
            first: tokens_1.color.white.base.value,
        },
        Input: {
            base: tokens_1.color.silver.base.value,
            focus: tokens_1.color.white_lilac.base.value,
            readonly: tokens_1.color.white.base.value,
        },
        InspectorPanel: {
            base: tokens_1.color.white.base.value,
            Icon: {
                base: tokens_1.color.white.base.value,
                hover: tokens_1.color.purple.base.value,
            },
        },
        ListCell: {
            Icon: {
                base: tokens_1.color.casper.base.value,
                highlight: tokens_1.color.purple.base.value,
            },
        },
        ListSubheader: {
            base: tokens_1.color.silver.base.value,
            first: tokens_1.color.teal.base.value,
            second: tokens_1.color.white.base.value,
        },
        Merchant: {
            settled: tokens_1.color.teal.base.value,
        },
        Modal: {
            base: tokens_1.color.white.base.value,
            overlay: tokens_1.color.charcoal.muted.value,
        },
        Tabs: {
            base: tokens_1.color.silver.base.value,
        },
        Tile: {
            base: tokens_1.color.mango.base.value,
        },
        Text: {
            light: tokens_1.color.white.base.value,
        },
        Status: {
            base: 'transparent',
            first: tokens_1.color.athens_gray.light.value,
            second: tokens_1.color.green.base.value,
            third: tokens_1.color.danger.base.value,
            fourth: tokens_1.color.mango.base.value,
        },
        Row: {
            base: tokens_1.color.athens_gray.base.value,
            first: tokens_1.color.silver.base.value,
            second: tokens_1.color.white.base.value,
        },
        SelectOption: {
            base: tokens_1.color.white.base.value,
            hover: tokens_1.color.silver.base.value,
        },
        Skeleton: {
            base: tokens_1.color.grey.base.value,
            highlight: tokens_1.color.grey.highlight.value,
        },
        Toast: {
            success: tokens_1.color.green.base.value,
            error: tokens_1.color.danger.base.value,
        },
        SelectorOption: {
            base: tokens_1.color.silver.base.value,
            active: tokens_1.color.purple.base.value,
        },
        Transaction: {
            Details: {
                base: tokens_1.color.athens_gray.base.value,
            },
        },
    },
    border: {
        seperator: tokens_1.color.black_haze.base.value,
        InformationBadge: {
            base: tokens_1.color.black_haze.base.value,
        },
        InspectorPanel: {
            header: tokens_1.color.black_haze.base.value,
        },
        SideBarNavigation: {
            base: tokens_1.color.black_haze.base.value,
        },
        NavigationItem: {
            base: tokens_1.color.black_haze.base.value,
        },
        Input: {
            base: tokens_1.color.manatee.base.value,
            focus: tokens_1.color.purple.base.value,
            readonly: tokens_1.color.silver.base.value,
        },
        Tabs: {
            base: tokens_1.color.mango.base.value,
        },
        Select: {
            base: tokens_1.color.purple.base.value,
            option: tokens_1.color.black_haze.base.value,
        },
        Transaction: {
            Details: {
                base: tokens_1.color.black_haze.base.value,
            },
        },
        Separator: {
            base: tokens_1.color.black_haze.base.value,
        },
    },
    PasswordPanel: {
        base: tokens_1.color.white.base.value,
        title: tokens_1.color.tuna.base.value,
        checkedTextColor: tokens_1.color.charcoal.base.value,
        defaultTextColor: tokens_1.color.grey.light.value,
    },
};
exports.default = exports.lightModeColors;
