// @fairfx/geometry-web/dist/themer/base/sizes.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const tokens = __importStar(require("../tokens"));
// @TODO ditch this and have them as named props
const sizes = [0, 1, 2, 4, 8, 10, 12, 16, 24, 32, 64, 96];
const small = sizes[6];
const base = sizes[7];
const times1p5 = base * 1.5;
const double = base * 2;
const times4 = base * 4;
const large = sizes[8];
const headerAction = 56;
const headerIcon = 18;
const actions = sizes[10];
const xLarge = 40; // TODO add this in sizes list
const sizesHeight = {
    actions,
    Button: {
        base: sizes[10],
    },
    Input: sizes[10],
    Select: {
        base: sizes[10],
        optionsOpen: 192,
    },
    RadioButton: {
        base: sizes[8],
    },
};
const sizesWidth = {
    BlockGroup: {
        fixed: tokens.grid.max_width.value + tokens.grid.outer_gutter.value * 2,
    },
    BlockItem: {
        max: tokens.grid.max_width.value + tokens.grid.inner_gutter.value * 2,
    },
    InspectorPanel: {
        max: 464,
    },
    Modal: {
        base: 550,
    },
    Toast: {
        max: 464,
    },
    Tooltip: {
        base: 140,
        border: sizes[5],
    },
    RadioButton: {
        base: sizes[8],
    },
};
const sizesSize = {
    Button: {
        Icon: {
            base: sizes[6],
        },
    },
    Icon: {
        small: sizes[6],
        action: sizes[10],
    },
    InformationBadge: {
        Icon: {
            base: 20,
        },
    },
    NavigationItem: {
        Icon: sizes[10],
    },
    Header: {
        Height: {
            base: '56px',
        },
    },
    ListHeader: {
        Height: {
            base: sizes[10],
            max: sizes[11],
        },
    },
    Dropdown: {
        base: sizes[8],
        small: sizes[7],
    },
    SelectorOption: {
        base: sizes[12],
    },
};
sizes.base = base;
sizes.small = small;
sizes.large = large;
sizes.xLarge = xLarge;
sizes.actions = actions;
sizes.headerAction = headerAction;
sizes.headerIcon = headerIcon;
sizes.width = sizesWidth;
sizes.height = sizesHeight;
sizes.size = sizesSize;
sizes.times4 = times4;
sizes.double = double;
sizes.times1p5 = times1p5;
exports.default = sizes;
