// @fairfx/geometry-web/dist/themer/base/space.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.space = void 0;
exports.space = [0, 1, 2, 4, 8, 10, 12, 16, 24, 32, 48, 64];
const base = exports.space[4];
const half = base / 2;
const double = base * 2;
const triple = base * 3;
const times4 = base * 4;
const none = exports.space[0];
const spaceMargin = {
    BlockGroup: {
        y: 24,
    },
    Block: {
        base: 0 - base,
    },
    Select: {
        base: {
            x: 14,
            y: 24,
        },
        active: {
            x: 10,
            y: 10,
        },
        chevron: {
            x: 27,
        },
    },
    Separator: {
        base: 16,
    },
    RadioButton: {
        bottom: exports.space[2],
        top: exports.space[7],
    },
};
const spacePadding = {
    BlockGroup: {
        x: 32,
    },
    BlockItem: {
        base,
    },
    Box: {
        base: 0,
        xsmall: base,
        small: 16,
        medium: 32,
        large: 48,
        xlarge: 64,
    },
    InformationBadge: {
        base: 5,
    },
    Inline: {
        none,
        small: exports.space[7],
        base: double,
    },
    Button: {
        base,
        Icon: {
            left: 10,
        },
    },
    InspectorPanelTitle: {
        x: 16,
    },
    NavigationItem: {
        Icon: 22,
        Text: {
            right: 64,
        },
    },
    Tile: {
        base: 16,
        Headline: {
            bottom: 36,
        },
    },
    Transaction: {
        details: {
            x: 16,
            y: 14,
        },
    },
    RadioButton: {
        left: exports.space[9],
    },
};
exports.space.base = base;
exports.space.double = double;
exports.space.half = half;
exports.space.triple = triple;
exports.space.times4 = times4;
exports.space.none = none;
exports.space.margin = spaceMargin;
exports.space.padding = spacePadding;
exports.default = exports.space;
