// @fairfx/geometry-web/dist/themer/base/typography.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.letterSpacings = exports.fonts = exports.lineHeights = exports.fontWeights = exports.fontSizes = void 0;
const tokens_1 = require("../tokens");
exports.fontSizes = [
    tokens_1.size.font.placeholder.value,
    tokens_1.size.font.cta.value,
    tokens_1.size.font.small.value,
    tokens_1.size.font.base.value,
    tokens_1.size.font.large.value,
    tokens_1.size.font.heading4.value,
    tokens_1.size.font.heading3.value,
    tokens_1.size.font.heading2.value,
    tokens_1.size.font.heading1.value,
    tokens_1.size.font.label.value,
];
exports.fontSizes.base = exports.fontSizes[3];
exports.fontSizes.body = exports.fontSizes.base;
exports.fontSizes.button = exports.fontSizes[1];
exports.fontSizes.cta = exports.fontSizes[1];
exports.fontSizes.placeholder = exports.fontSizes[0];
exports.fontSizes.small = exports.fontSizes[2];
exports.fontSizes.large = exports.fontSizes[4];
exports.fontSizes.label = exports.fontSizes[9];
exports.fontSizes.h2 = exports.fontSizes[7];
exports.fontSizes.h4 = exports.fontSizes[5];
exports.fontWeights = {
    body: tokens_1.weight.font.body.value,
    heading: tokens_1.weight.font.heading.value,
    semi: tokens_1.weight.font.semi.value,
    bold: tokens_1.weight.font.bold.value,
    regular: tokens_1.weight.font.regular.value,
    medium: tokens_1.weight.font.medium.value,
};
exports.lineHeights = {
    body: tokens_1.line_height.font.body.value,
    heading: tokens_1.line_height.font.heading.value,
    button: tokens_1.line_height.font.cta.value,
    cta: tokens_1.line_height.font.cta.value,
};
exports.fonts = {
    body: 'Equals, sans-serif',
    heading: 'Equals, sans-serif',
};
exports.letterSpacings = {
    base: '-0.5',
};
exports.default = { fonts: exports.fonts, lineHeights: exports.lineHeights, fontWeights: exports.fontWeights, fontSizes: exports.fontSizes, letterSpacings: exports.letterSpacings };
