// @fairfx/geometry-web/dist/components/toast-manager/Toast.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToastMessage = exports.Toast = void 0;
const core_1 = require("@emotion/core");
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
const commonWidth = '464px';
exports.Toast = theme_1.default.div `
position: fixed;
right: 0;
bottom: 0;
maxWidth: ${commonWidth},
width: 100%;
transition: opacity 300ms ease-in-out;
opacity: 1;
  ${(props) => ({
    entering: core_1.css `
        opacity: 1;
      `,
    entered: core_1.css `
        opacity: 1;
      `,
}[props.stage])}
  },
`;
exports.ToastMessage = theme_1.default.div `
border-radius: 2;
width: ${commonWidth};
cursor: pointer;
padding: ${tokens_1.sizes[4]}px ${tokens_1.sizes[8]}px;
min-height: actions;
text-align: left;
  ${(props) => ({
    success: core_1.css `
        background: ${tokens_1.color.green.base.value};
      `,
    error: core_1.css `
        background: ${tokens_1.color.danger.base.value};
      `,
    default: core_1.css `
        background: transparent;
      `,
}[props.type])}
  },
`;
