// @fairfx/geometry-palette/src/index.ts
export const flatColorPalette = {
  // Mango
  'mango.0': '#FFB900',
  'mango.1': '#FFD566',
  // Midnight Blue
  'midnightBlue.0': '#262835',
  // Sky Blue
  'skyBlue.0': '#F7FAFD',
  'skyBlue.1': '#E5F2FC',
  'skyBlue.2': '#C0DEF8',
  'skyBlue.3': '#99CAF4',
  'skyBlue.4': '#75B6EF',
  'skyBlue.5': '#5DA6EC',
  'skyBlue.6': '#4C97E8',
  'skyBlue.7': '#4689DA',
  'skyBlue.8': '#3871C2',
  'skyBlue.9': '#3260B0',
  // Sea Green
  'seaGreen.0': '#D5F1F3',
  'seaGreen.1': '#C1EAED',
  'seaGreen.2': '#ACE3E7',
  'seaGreen.3': '#9CDEE1',
  'seaGreen.4': '#82D5DA',
  'seaGreen.5': '#6DCDD3',
  'seaGreen.6': '#59C7CE',
  'seaGreen.7': '#2FB9C2',
  'seaGreen.8': '#238B92',
  'seaGreen.9': '#185D61',
  // Grey
  'grey.0': '#F7F8FC',
  'grey.1': '#EDEDF4',
  'grey.2': '#E1E1EA',
  'grey.3': '#D4D5DD',
  'grey.4': '#BCBCC9',
  'grey.5': '#9699AA',
  'grey.6': '#7C7E8D',
  'grey.7': '#676978',
  'grey.8': '#484A58',
  //   Red
  'red.0': '#FFF2F1',
  'red.1': '#D95852',
  // Yellow
  'yellow.0': '#FFF8E4',
  // Green
  'green.0': '#0BA866',
  // White
  'white.0': '#FFFFFF',
};

export const nestedColorPalette = {
  mango: ['#FFB900', '#FFD566'],
  midnightBlue: ['#262835'],
  skyBlue: [
    '#F7FAFD',
    '#E5F2FC',
    '#C0DEF8',
    '#99CAF4',
    '#75B6EF',
    '#5DA6EC',
    '#4C97E8',
    '#4689DA',
    '#3871C2',
    '#3260B0',
  ],
  seaGreen: [
    '#D5F1F3',
    '#C1EAED',
    '#ACE3E7',
    '#9CDEE1',
    '#82D5DA',
    '#6DCDD3',
    '#59C7CE',
    '#2FB9C2',
    '#238B92',
    '#185D61',
  ],
  grey: [
    '#F7F8FC',
    '#EDEDF4',
    '#E1E1EA',
    '#D4D5DD',
    '#BCBCC9',
    '#9699AA',
    '#7C7E8D',
    '#676978',
    '#484A58',
  ],
  red: ['#FFF2F1', '#D95852'],
  yellow: ['#FFF8E4'],
  green: ['#0BA866'],
  white: ['#FFFFFF'],
};

export const colorPalette = {
  ...nestedColorPalette,
  ...flatColorPalette,
};
