// @fairfx/geometry-web/dist/components/spinner/Spinner.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Wrapper = exports.Inner = exports.Outer = void 0;
const theme_1 = __importDefault(require("../../theme"));
exports.Outer = theme_1.default.div `
  display: inline-block;
  position: relative;
  width: ${p => p.width};
  height: ${p => p.height};
`;
exports.Inner = theme_1.default.div `
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  border-width: ${p => p.width};
  border-style: solid;
  border-radius: 50%;
  border-color: ${p => p.color ? p.color : p.theme.context.colors.spinner.stroke}
    transparent transparent transparent;
  animation: ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;

  &:nth-of-type(1) {
    animation-delay: -0.45s;
  }

  &:nth-of-type(2) {
    animation-delay: -0.3s;
  }

  &:nth-of-type(3) {
    animation-delay: -0.15s;
  }

  @keyframes ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;
exports.Wrapper = theme_1.default.div `
  text-align: center;
  width: ${p => p.width};
  height: ${p => p.height};
`;
