// @fairfx/geometry-web/dist/components/_deprecatedModal/Modal.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModalBody = exports.ModalInformation = exports.ModalKey = exports.SubheaderContainer = exports.ModalSubheader = exports.ActionContainer = exports.HeaderWrapper = exports.TextWrapper = exports.ModalHeader = exports.ModalContent = exports.ModalContainer = void 0;
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
exports.ModalContainer = theme_1.default.div `
  display: flex;
  align-items: center;
  position: fixed;
  right: 0;
  top: 0;
  left: 0;
  bottom: 0;
  z-index: ${tokens_1.indices.modal};
  background-color: ${tokens_1.color.charcoal.muted.value};
`;
exports.ModalContent = theme_1.default.div `
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  min-height: 240px;
  opacity: 1;
  right: 30%;
  position: fixed;
  background-color: ${tokens_1.color.white.base.value};
  width: 550px;
  margin: 0 auto;
`;
exports.ModalHeader = theme_1.default.span `
  font-size: ${tokens_1.sizes[4] * 3}px;
  font-weight: bold;
  line-height: ${tokens_1.sizes[9]}px;
  margin: 0 auto;
  &::first-letter {
    text-transform: capitalize;
  }
`;
exports.TextWrapper = theme_1.default.div `
  padding: 0 ${tokens_1.sizes[9]}px ${tokens_1.sizes[9]}px;
  margin-bottom: ${tokens_1.sizes[5]}px;
`;
exports.HeaderWrapper = theme_1.default.div `
  display: flex;
  flex-direction: column;
  padding-top: ${tokens_1.sizes[9]}px;
`;
exports.ActionContainer = theme_1.default.div `
  position: fixed;
  bottom: 0;
  right: 0;
  left: 0;
  width: 100%;
`;
exports.ModalSubheader = theme_1.default.span `
  font-size: ${tokens_1.sizes[7]}px;
  line-height: ${tokens_1.sizes[9]}px;
`;
exports.SubheaderContainer = theme_1.default.div `
  padding: ${tokens_1.sizes[7]}px 0;
`;
exports.ModalKey = theme_1.default.span `
  font-weight: bold;
  &::first-letter {
    text-transform: capitalize;
  }
`;
exports.ModalInformation = theme_1.default.div `
  margin-top: ${tokens_1.sizes[3]}px;
`;
exports.ModalBody = theme_1.default.div `
  font-size: ${tokens_1.sizes[4] * 3}px;
`;
