// @fairfx/geometry-web/dist/components/tabs/Tab.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TabHeader = exports.TabAnchor = exports.Tab = void 0;
const core_1 = require("@emotion/core");
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
exports.Tab = theme_1.default.div `
display: none;
  ${(props) => props.active
    ? core_1.css `
          padding: ${tokens_1.sizes[7]}px ${tokens_1.sizes[8]}px ${tokens_1.sizes[10] + tokens_1.sizes[8]}px;
          display: block;
          height: 100%;
        `
    : ''}
  }
`;
exports.TabAnchor = theme_1.default.a `
  color: inherit;
  text-decoration: inherit;
`;
exports.TabHeader = theme_1.default.li `
  display: flex;
  align-items: center;
  font-size: ${tokens_1.fontSizes[2]}px;
  cursor: pointer;
  padding-left: ${tokens_1.sizes[7]}px;
  padding-right: ${tokens_1.sizes[7]}px;
  margin-left: ${tokens_1.sizes[4]}px;
  margin-right: ${tokens_1.sizes[4]}px;
  margin-bottom: 0;
  ${(props) => props.active
    ? core_1.css `
          font-weight: ${tokens_1.weight.font.bold.value};
          box-shadow: inset ${tokens_1.color.mango.base.value} 0px -4px 0px 0px;
        `
    : ''}
  },
`;
