// @fairfx/geometry-web/dist/components/tabs/Tabs.components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TabsBody = exports.TabsHeader = exports.TabsContainer = void 0;
const theme_1 = __importDefault(require("../../theme"));
const tokens_1 = require("../../themer/tokens");
exports.TabsContainer = theme_1.default.div `
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  height: 100%;
`;
exports.TabsHeader = theme_1.default.ul `
  display: flex;
  background: ${tokens_1.color.white.base.value};
  min-height: ${tokens_1.sizes[10]}px;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: ${tokens_1.color.black_haze.base.value};
  width: 100%;
  padding: 0;
  margin: 0;
`;
exports.TabsBody = theme_1.default.div `
  width: 100%;
  flex-grow: 1;
  over-flow: auto;
  max-height: calc(100% - ${tokens_1.sizes[10]}px);
  overflow: auto;
`;
