// @fairfx/geometry-web/dist/themer/tokens.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.indices = exports.fontSizes = exports.spaces = exports.sizes = exports.grid = exports.breakpoints = exports.transition = exports.line_height = exports.weight = exports.size = exports.color = void 0;
/* eslint-disable @typescript-eslint/camelcase */
const rgba_1 = __importDefault(require("polished/lib/color/rgba"));
exports.color = {
    athens_gray: {
        base: { value: '#f2f3f7' },
        light: { value: '#8C8C97' },
    },
    black_haze: {
        base: { value: '#E3E6E6' },
    },
    casper: {
        dark: { value: '#9CACBF' },
        base: { value: '#ABBCD1' },
    },
    charcoal: {
        base: { value: '#262835' },
        light: { value: '#2b2e3d' },
        muted: { value: rgba_1.default('#262835', 0.5) },
    },
    dandelion: {
        base: { value: '#FFD566' },
    },
    gallery: {
        base: { value: '#F0F0F0' },
    },
    green: {
        base: { value: '#1EDC8C' },
    },
    blue: {
        10: { value: '#C0DEF8' },
        20: { value: '#F7FAFD' },
        30: { value: '#E5F2FC' },
        40: { value: '#C0DEF8' },
        50: { value: '#99CAF4' },
        60: { value: '#75B6EF' },
        70: { value: '#5DA6EC' },
        80: { value: '#4689DA' },
        90: { value: '' },
        100: { value: '#3260B0' },
        110: { value: '#3871C2' },
    },
    manatee: {
        base: { value: '#8C8C97' },
    },
    mango: {
        base: { value: '#FFB900' },
    },
    midnight: {
        base: { value: '#00192E' },
    },
    mulled_wine: {
        base: { value: '#484A67' },
    },
    purple: {
        dark: { value: '#5E5E88' },
        base: { value: '#6C6C9B' },
        muted: { value: rgba_1.default('#6C6C9B', 0.5) },
        light: { value: '#C4C3D7' },
    },
    danger: {
        base: { value: '#F5635D' },
        dark: { value: '#D95852' },
        light: { value: '#FFF2F1' },
    },
    silver: {
        base: { value: '#F5F5F4' },
    },
    teal: {
        base: { value: '#00828E' },
    },
    waterloo: {
        base: { value: '#848692' },
    },
    white: {
        base: { value: '#FFFFFF' },
    },
    white_lilac: {
        base: { value: '#F7F7FC' },
    },
    grey: {
        base: { value: '#E4E5EA' },
        highlight: { value: '#EEEEEE' },
        light: { value: '#9699AA' },
        dark: { value: '#7C7E8D' },
        mutated: { value: '#BCBCC9' },
        mutatedLight: { value: '#EDEDF4' },
    },
    mid_gray: {
        base: { value: '#676978' },
    },
    lighter: {
        base: { value: '#757575' },
    },
    tuna: {
        base: {
            value: '#303344',
        },
    },
    mischka: {
        base: { value: '#E1E1EA' },
    },
    boxShadow: {
        base: { value: '#00000021' },
    },
};
exports.size = {
    base: { value: 16 },
    font: {
        placeholder: { value: 10 },
        label: { value: 12 },
        cta: { value: 12 },
        small: { value: 14 },
        base: { value: 16 },
        large: { value: 18 },
        // @todo can we give these more meaningful names
        heading4: { value: 24 },
        heading3: { value: 32 },
        heading2: { value: 46 },
        heading1: { value: 64 },
    },
};
exports.weight = {
    font: {
        body: { value: 400 },
        heading: { value: 700 },
        semi: { value: 500 },
        bold: { value: 700 },
        regular: { value: 400 },
        medium: { value: 600 },
    },
};
exports.line_height = {
    font: {
        body: { value: 1.35 },
        heading: { value: 1.2 },
        cta: { value: 1 },
    },
};
exports.transition = {
    base: { value: '300ms' },
    fast: { value: '200ms' },
};
exports.breakpoints = {
    large_phone: { value: '50rem' },
    tablet: { value: '66rem' },
    super_wide: { value: '107rem' },
};
exports.grid = {
    outer_gutter: { value: 32 },
    inner_gutter: { value: 8 },
    max_width: { value: 1000 },
};
// @TODO ditch this and have them as named props
exports.sizes = [0, 1, 2, 4, 8, 10, 12, 16, 24, 32, 64, 96];
exports.spaces = [0, 1, 2, 4, 8, 10, 12, 16, 24, 32, 48, 64];
exports.fontSizes = [
    exports.size.font.placeholder.value,
    exports.size.font.cta.value,
    exports.size.font.small.value,
    exports.size.font.base.value,
    exports.size.font.large.value,
    exports.size.font.heading4.value,
    exports.size.font.heading3.value,
    exports.size.font.heading2.value,
    exports.size.font.heading1.value,
    exports.size.font.label.value,
];
exports.indices = {
    select: 700,
    tooltip: 600,
    modal: 500,
    inspectorPanel: 400,
    inspectorPanelMobile: 400,
    sideBarNavigation: 300,
    header: 200,
    listSubheader: 100,
};
exports.default = {
    grid: exports.grid,
    breakpoints: exports.breakpoints,
    transition: exports.transition,
    line_height: exports.line_height,
    weight: exports.weight,
    size: exports.size,
    color: exports.color,
    sizes: exports.sizes,
    indices: exports.indices,
};
