// @fairfx/geometry-web/dist/components/sideBarNavigation/SideBarNavigation.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sideBarNavigationEquals = exports.sideBarNavigationFFX = void 0;
const geometry_palette_1 = require("@fairfx/geometry-palette");
exports.sideBarNavigationFFX = {
    background: geometry_palette_1.colorPalette.grey[0],
    actions: {
        text: {
            default: geometry_palette_1.colorPalette.midnightBlue[0],
            focused: geometry_palette_1.colorPalette.skyBlue[8],
            error: geometry_palette_1.colorPalette.midnightBlue[0],
            complete: geometry_palette_1.colorPalette.midnightBlue[0],
            hover: geometry_palette_1.colorPalette.skyBlue[7],
        },
        icon: {
            default: geometry_palette_1.colorPalette.midnightBlue[0],
            focused: geometry_palette_1.colorPalette.skyBlue[8],
            error: geometry_palette_1.colorPalette.midnightBlue[0],
            complete: geometry_palette_1.colorPalette.midnightBlue[0],
            hover: geometry_palette_1.colorPalette.skyBlue[7],
        },
        border: {
            default: geometry_palette_1.colorPalette.grey[0],
            focused: geometry_palette_1.colorPalette.skyBlue[8],
            error: geometry_palette_1.colorPalette.grey[0],
            complete: geometry_palette_1.colorPalette.grey[0],
            hover: geometry_palette_1.colorPalette.skyBlue[7],
        },
        background: {
            default: geometry_palette_1.colorPalette.grey[0],
            focused: geometry_palette_1.colorPalette.white[0],
            error: geometry_palette_1.colorPalette.grey[0],
            complete: geometry_palette_1.colorPalette.grey[0],
        },
    },
};
exports.sideBarNavigationEquals = {
    background: geometry_palette_1.colorPalette.grey[8],
    actions: {
        text: {
            default: geometry_palette_1.colorPalette.white[0],
            focused: geometry_palette_1.colorPalette.white[0],
            error: geometry_palette_1.colorPalette.white[0],
            complete: geometry_palette_1.colorPalette.white[0],
            hover: geometry_palette_1.colorPalette.mango[0],
        },
        icon: {
            default: geometry_palette_1.colorPalette.white[0],
            focused: geometry_palette_1.colorPalette.mango[0],
            error: geometry_palette_1.colorPalette.white[0],
            complete: geometry_palette_1.colorPalette.white[0],
            hover: geometry_palette_1.colorPalette.mango[0],
        },
        border: {
            default: geometry_palette_1.colorPalette.white[0],
            focused: geometry_palette_1.colorPalette.mango[0],
            error: geometry_palette_1.colorPalette.white[0],
            complete: geometry_palette_1.colorPalette.white[0],
            hover: geometry_palette_1.colorPalette.mango[0],
        },
        background: {
            default: geometry_palette_1.colorPalette.grey[8],
            focused: geometry_palette_1.colorPalette.midnightBlue[0],
            error: geometry_palette_1.colorPalette.grey[8],
            complete: geometry_palette_1.colorPalette.grey[8],
            hover: geometry_palette_1.colorPalette.grey[8],
        },
    },
};
